clar2wasm/src/wasm_generator.rs

```diff
@@ -1010,13 +1008,655 @@ impl WasmGenerator {
         size
     }
 
+    /// Serialize the value of type `ty` on the top of the data stack using
+    /// consensus serialization. Leaves the length of the data written on the
+    /// top of the data stack.
+    fn serialize_to_memory(
+        &mut self,
+        builder: &mut InstrSeqBuilder,
+        offset_local: LocalId,
+        offset: u32,
+        ty: &TypeSignature,
+    ) {
+        let memory = self
+            .module
+            .memories
+            .iter()
+            .next()
+            .expect("no memory found")
+            .id();
+
+        use clarity::vm::types::signatures::TypeSignature::*;
+        match ty {
```

Comment:
Maybe each branch could have some kind of high level explanation of what should happen in the generated Wasm?
Because even if I'm used to write Wasm now, I still needed a few times reading this to understand where it was going.

Reply:
Done in eee42fc.